caddyhttp/proxy/proxy.go

```diff
@@ -197,10 +218,10 @@ func (p Proxy) ServeHTTP(w http.ResponseWriter, r *http.Request) (int, error) {
 		timeout := host.FailTimeout
 		if timeout > 0 {
 			atomic.AddInt32(&host.Fails, 1)
-			go func(host *UpstreamHost, timeout time.Duration) {
+			go func() {
 				time.Sleep(timeout)
 				atomic.AddInt32(&host.Fails, -1)
-			}(host, timeout)
+			}()
```

Comment:
Wait a sec -- why are these parameters removed? It is a bug to start a goroutine in a `for` loop and not pass in the values as they are on this iteration of the loop, they might change before the goroutine starts.